src/System.Private.ServiceModel/tests/Scenarios/Binding/Custom/CustomBindingTests.cs

```diff
@@ -51,4 +51,69 @@ public static void DefaultSettings_Tcp_Binary_Echo_RoundTrips_String()
             ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
         }
     }
+
+    public class MessageModifier : ChannelMessageInterceptor
+    {
+        int send = 0;
+
+        public override void OnSend(ref Message msg)
+        {
+            send++;
+            if (send % 2 == 0)
+            {
+                // Add extra header so that when the message arrives on service side, the message won't be dropped
+                msg.Headers.Add(MessageHeader.CreateHeader("ByPass", "urn:InterceptorNamespace", "ByPassPassword"));
+            }
+        }
+
+        public override ChannelMessageInterceptor Clone()
+        {
+            return new MessageModifier();
+        }
+    }
+
+    [Fact]
+    [OuterLoop]
+    [ActiveIssue(1170)]
+    public static void CustomBinding_Message_Interceptor()
+    {
+        ChannelFactory<IChannelExtensibilityContract> factory = null;
+        IChannelExtensibilityContract serviceProxy = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            CustomBinding binding = new CustomBinding(
+                new InterceptingBindingElement(new MessageModifier()),
+                new HttpTransportBindingElement());
+
+            factory = new ChannelFactory<IChannelExtensibilityContract>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_ChannelExtensibility));
+            serviceProxy = factory.CreateChannel();
```

Comment:
You opened #1170 and marked this test ActiveIssue with it.  Did you investigate why it failed?  Does it pass if you call factory.Open() first?